lib/secrets.rb

```diff
@@ -189,6 +237,17 @@ def secrets_from_annotations(annotations)
     end.compact
   end
 
+  # pki/FOO="a/b/c/z" -> {"FOO" => "a/b/c/d"}
+  def pki_from_annotations(annotations)
+    # TODO: this function is too similar to secrets_from_annotations
```

Comment:
I can do the refactor once the PR is otherwise stable ... needs to not read annotations twice too :)

Reply:
Took a shot at this refactor. If its not what you'd like, I can easily revert.